packages/starknet-snap/src/rpcs/estimateFee.ts

```diff
@@ -0,0 +1,151 @@
+   * @param params.transactions - The list of transactions to be signed. Reference: https://www.starknetjs.com/docs/API/namespaces/types#call
+   * @param params.transactionsDetail - The InvocationsSignerDetails of the transactions to be signed. Reference: https://www.starknetjs.com/docs/API/namespaces/types#invocationssignerdetails
+   * @param [params.enableAuthorize] - Optional, a flag to enable or bypass the confirmation dialog.
+   * @returns the signature of the transaction in a string array.
+   */
+  async execute(params: EstimateFeeParams): Promise<EstimateFeeResponse> {
+    return super.execute(params);
+  }
+
+  protected async handleRequest(
+    params: EstimateFeeParams,
+  ): Promise<EstimateFeeResponse> {
+    const {
+      address,
+      contractAddress,
+      contractFuncName,
+      contractCallData,
+      transactionVersion,
+    } = params;
+    const txnInvocation = {
+      contractAddress,
+      entrypoint: contractFuncName,
+      calldata: contractCallData.split(',').map((ele) => ele.trim()),
+    };
+    const accountDeployed = await isAccountDeployed(this.network, address);
+
+    logger.log(`estimateFee:\ntxnInvocation: ${toJson(txnInvocation)}`);
+
+    // Estimate deploy account fee if the signer has not been deployed yet
+    let bulkTransactions: Invocations = [
+      {
+        type: TransactionType.INVOKE,
+        payload: txnInvocation,
+      },
+    ];
+
+    if (!accountDeployed) {
+      const { callData } = getAccContractAddressAndCallData(
+        this.account.publicKey,
+      );
+      const deployAccountpayload = {
+        classHash: ACCOUNT_CLASS_HASH,
+        contractAddress: address,
+        constructorCalldata: callData,
+        addressSalt: this.account.publicKey,
+      };
+
+      bulkTransactions = [
+        {
+          type: TransactionType.DEPLOY_ACCOUNT,
+          payload: deployAccountpayload,
+        },
+        {
+          type: TransactionType.INVOKE,
+          payload: txnInvocation,
+        },
+      ];
+    }
+
+    const estimateBulkFeeResp = await estimateFeeBulk(
+      this.network,
+      address,
+      this.account.privateKey,
+      bulkTransactions,
+      transactionVersion as '0x2' | '0x3',
+    );
+    logger.log(
+      `estimateFee:\nestimateFeeBulk estimateBulkFeeResp: ${toJson(
+        estimateBulkFeeResp,
+      )}`,
+    );
+    const estimateFeeResp = addFeesFromAllTransactions(estimateBulkFeeResp);
+
+    logger.log(`estimateFee:\nestimateFeeResp: ${toJson(estimateFeeResp)}`);
+
+    const resp = {
+      suggestedMaxFee: estimateFeeResp.suggestedMaxFee.toString(10),
+      overallFee: estimateFeeResp.overall_fee.toString(10),
+      unit: 'wei',
+      includeDeploy: !accountDeployed,
+    };
+    logger.log(`estimateFee:\nresp: ${toJson(resp)}`);
+
+    return resp;
+  }
+}
+
+export const estimateFee = new EstimateFeeRpc({
+  showInvalidAccountAlert: false,
+  checkDeploy: false,
+  checkUpgrade: true,
```

Comment:
we dont need that , we should not reuse the api in other api
```suggestion
```

Reply:
Please have a look at #330 . Because in estimateFee we estimateFee together with deploy if account not deployed. But current rpc handler will throw error if deploy is not done, this is why we need to be able in some cases to skip the checks